### PR TITLE
Symantec mss integration

### PR DESCRIPTION
**Pull request by guydemi**
Still not ready, work in progress.

### CODE REVIEW

File: Integrations/integration-SymantecMSS.yml [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+            # For each recent event - query for additional details
+            dictQuery = queryIncident(dictIncident['IncidentNumber'])
+
+            # Merge both dicts
+            element.update(dictIncident)
+            element.update(dictQuery)
+            result.append(element)
+
+            ### CONSIDER SLEEPING HERE IN CASE WE GET BLOCKED BY THE SERVER
```
Comment (dorsha) [inline Integrations/integration-SymantecMSS.yml]: Try not pushing TODOs, it is bad practice.

**Comment by anara123** [comment]
resolves https://github.com/demisto/etc/issues/10076

**Comment by guydemi** [comment]
# Documentation:

### Overview:
An integration with Symantec Managed Security Services.
The integration provides the ability create Demisto incidents from Symantec incidents.

### To set up syslog to work with Demisto:
1. Export a Production certificate that enables you to access your organization’s information in
SWS (https://api.monitoredsecurity.com/SWS/) in .p12 format.
2. Use any "File to Base64" convertor to encode the .p12 file into a base64 string.

### To set up the integration on Demisto:
1. Go to ‘Settings > Integrations > Servers & Services’
2. Locate Symantec MSS by searching for it using the search box on the top of the page.
3. Click ‘Add instance’ to create and configure a new integration. You should configure the following settings:  

   **Name**: A textual name for the integration instance.

   **Certificate Passphrase**: The passphrase used to create the .p12 certificate.

   **Certificate**: The base64 representation of the exported production certificate (as explained above).

   **Test** button checks that the certificate is authenticated and the SWS server is responsive.

### Fetched incidents data:
Incidents with the severities of "Emergency" or "Critical" will be fetched.
When importing events for the first time, incidents from the last 10 minutes are imported.
A maximum of 100 incidents will be created in one import.

### Use-case examples:
* Close an incident
Example: "!symantec-mss-update-incident number=123"

* Query for a specific incident
Example: "!symantec-mss-get-incident number=123"

* Retreive a list of alerts and them as incidents into Demisto
Check "Import events as incidents" when configuring the integration.
To get a list of incidents from the War Room: "!symantec-mss-incidents-list"

### Commands:
* "!symantec-mss-incidents-list": Get a list of incidents, you can add specific filters like time, source ip, severities and max incidents. If no time is supplied, incidents from the last 24 hours are returned.

* "!symantec-mss-get-incident": Query an incident by number.

* "!symantec-mss-update-incident": Updates an incident's workflow by incident number. Any optional parameter that isn't supplied, is taken from the current workflow, in case non exist, "Resolution" is defaulted to "Resolved", "Status" is defaulted to "Closed" and "Severity" defaults to "Informational".

### Known Limitations

### Troubleshooting 
* If unauthorized, make sure the exported .p12 certificate is for the production API and not the test API
* Make sure that https requests can be made from your machine
* Make sure to use the correct proxy and enable it in the configuration if needed

**Comment by dorsha** [comment]
@mayairv above you have integration documentation

File: Integrations/integration-SymantecMSS.yml [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+            f_pem.write(OpenSSL.crypto.dump_certificate(OpenSSL.crypto.FILETYPE_PEM, p12.get_certificate()))
+            ca = p12.get_ca_certificates()
+            if ca is not None:
+                for cert in ca:
+                    f_pem.write(OpenSSL.crypto.dump_certificate(OpenSSL.crypto.FILETYPE_PEM, cert))
+            f_pem.close()
+            yield t_pem.name
+
+
+    def apiCall(body, headers):
```
Comment (anara123) [inline Integrations/integration-SymantecMSS.yml]: apiCall => api_call

File: Integrations/integration-SymantecMSS.yml [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+        events = getIncidentsListRequest(time=lastRun, srcIp=None, severities="Emergency,Critical", maxIncidents=100)
+        for event in events:
+            inc = event_to_incident(event)
+            incidents.append(inc)
+
+        demisto.incidents(incidents)
+        demisto.setLastRun({'time': now})
+
+
+    def getIncidentsList(time):
```
Comment (anara123) [inline Integrations/integration-SymantecMSS.yml]: get_incident_list

File: Integrations/integration-SymantecMSS.yml [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+            'ContentsFormat': formats['json'],
+            'Type': entryTypes['note'],
+            'Contents': result,
+            'EntryContext': context,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': hr
+        })
+
+
+    def getIncidentsListRequest(time, srcIp, severities, maxIncidents):
```
Comment (anara123) [inline Integrations/integration-SymantecMSS.yml]: get_incident_list_request

File: Integrations/integration-SymantecMSS.yml [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+        result = []
+        for incident in incidentNodes:
+            stringIncidentXml = xml.etree.ElementTree.tostring(incident)
+            stringIncidentJson = xml2json(stringIncidentXml)
+            dictIncident = json.loads(stringIncidentJson)['SecurityIncidentSummary']
+            result.append(dictIncident)
+        return result
+
+
+    def updateIncident():
```
Comment (anara123) [inline Integrations/integration-SymantecMSS.yml]: update_incident
change all function names like that

File: Integrations/integration-SymantecMSS.yml [hunk 6]
```diff
@@ -0,0 +1,506 @@
...
+            "AnalystAssessment": result["AnalystAssessment"],
+            "CountryCode": result["CountryCode"],
+            "CountryName": result["CountryName"],
+            "RelatedTickets": result["RelatedTickets"],
+            "WorkFlowDetail": {
+                "Status": result["WorkFlowDetail"]["Status"],
+                "AssignedPerson": result["WorkFlowDetail"]["AssignedPerson"]
+            },
+            "IncidentComments": {
+                "IncidentComment": { # array?
```
Comment (anara123) [inline Integrations/integration-SymantecMSS.yml]: what is this comment?

File: Integrations/integration-SymantecMSS.yml [hunk 7]
```diff
@@ -0,0 +1,506 @@
...
+                "IncidentAttachmentItem": { # array?
+                    "AttachmentNumber": result["IncidentAttachmentItems"]["IncidentAttachmentItem"]["AttachmentNumber"],
+                    "AttachmentName": result["IncidentAttachmentItems"]["IncidentAttachmentItem"]["AttachmentName"],
+                    "UploadDateGMT": result["IncidentAttachmentItems"]["IncidentAttachmentItem"]["UploadDateGMT"],
+                    "UploadBy": result["IncidentAttachmentItems"]["IncidentAttachmentItem"]["UploadBy"],
+                    "Comment": result["IncidentAttachmentItems"]["IncidentAttachmentItem"]["Comment"]
+                }
+            },
+            "RelatedIncidents": {
+                "IncidentNumber": result["RelatedIncidents"]["IncidentNumber"] # array for sure
```
Comment (anara123) [inline Integrations/integration-SymantecMSS.yml]: if you don't sure, you always can use `dq`. `dq(result, 'RelatedIncidents. IncidentNumber')`

**Comment by anara123** [comment]
Looks good overall. Just fix the code review notes and add test playbook

**Comment by anara123** [comment]
@guydemi for docs, please open new issue and put there the docs and assign it to @mayairv

File: Integrations/integration-SymantecMSS.yml [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+  threat monitoring and customized guidance 24x7
+detaileddescription: Encode the .p12 certificate file using any File to Base64 Encoder
+  and paste the encoded string in the "Certificate" field
+configuration:
+- display: Server URL
+  name: server
+  defaultvalue: https://api.monitoredsecurity.com
+  type: 0
+  required: true
+- display: Certificate (Base64 encoded)
```
[the remark this answers is not in the document]
Comment (guydemi) [reply Integrations/integration-SymantecMSS.yml]: explained in docs & detailed description,
as discussed pem is redundant as the export is in p12 format.
adding .pem support  in the future will be very low effort if ever needed as the integration supports it natively,

**Comment by anara123** [comment]
@guydemi please add test playbook

**Pull request merged**